blocks/aem-asset-selector/aem-asset-selector-util.js

```diff
@@ -55,62 +155,215 @@ export function init(cfg, callback) {
+ * @returns {Promise} Resolves when the item has been written to the clipboard.
+ */
+async function copyToClipboardWithBinary(assetPublicUrl, mimeType) {
+  const binary = await fetch(assetPublicUrl);
+
+  if (!binary.ok) {
+    throw new Error(`Unexpected status code ${binary.status} retrieving asset binary`);
+  }
+
+  const blob = await binary.blob();
+  const clipboardOptions = {};
+  clipboardOptions[mimeType] = blob;
+  const data = [
+    new ClipboardItem(clipboardOptions),
+  ];
+  return navigator.clipboard.write(data);
 }
 
+/**
+ * Copies the given asset to the clipboard, without using the Repository API,
+ * and therefore avoiding calls directly to AEM. The primary case for using
+ * this method would be if AEM's CORS policies would deny requests from the
+ * asset selector.
+ * @param {Asset} asset Asset repository metadata, as provided by the asset
+ *  selector.
+ * @returns {Promise<boolean>} Resolves with true if the asset was
+ *  copied successfully, otherwise resolves with false.
+ */
+export async function copyAssetWithoutRapi(asset) {
+  const maxRendition = getCopyRendition(asset);
+  if (!maxRendition) {
+    logMessage('No suitable rendition to copy found');
+    return false;
+  }
+  try {
+    const assetPublicUrl = await getAssetPublicUrl(maxRendition.href.substring(0, maxRendition.href.indexOf('?')));
+    if (!assetPublicUrl) {
+      logMessage('Unable to generate public url for copy');
+      return false;
+    }
+    await copyToClipboardWithHtml(assetPublicUrl);
+  } catch (e) {
+    logMessage('error copying asset to clipboard', e);
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Copies the given asset to the clipboard, using the Repository API to
+ * generate a download URL for the asset, then using the generated URL
+ * to copy the asset's content to the clipboard.
+ * @param {Asset} asset Asset repository metadata, as provided by the
+ *  asset selector.
+ * @returns {Promise<boolean>} Resolves with true if the asset was
+ *  copied successfully, otherwise resolves with false.
+ */
+export async function copyAssetWithRapi(asset) {
+  // eslint-disable-next-line no-underscore-dangle
+  if (!asset) {
+    logMessage('Asset metadata does not contain sufficient information');
+    return false;
+  }
+  const rendition = getCopyRendition(asset);
+  if (!rendition) {
+    logMessage('No suitable rendition to copy found');
+    return false;
+  }
+  const download = getRel(rendition, REL_DOWNLOAD);
+  if (!download || !download.href) {
+    logMessage('Rendition does not contain sufficient information');
+    return false;
+  }
+  try {
+    const url = download.href;
+    const res = await fetch(url, {
+      headers: {
+        Authorization: `Bearer ${imsInstance.getImsToken()}`,
+      },
+    });
+    if (!res.ok) {
+      logMessage(`Download request for rendition binary returned unexpected status code ${res.status}: ${res.statusText}`);
```

Comment:
```suggestion
      logMessage(`Download request for rendition binary failed with status code ${res.status}: ${res.statusText}`);
```
If it's unexpected, it means it failed for our use case, right?